src/services/copyConfig/rba/rba.js

```diff
@@ -84,14 +91,39 @@ export default class Rba {
     return response
   }
 
-  async setRbaRulesAndSettings(destinationSite, destinationSiteConfiguration, payload) {
-    const response = await this.#rbaPolicy.set(destinationSite, payload, destinationSiteConfiguration.dataCenter)
+  // Merge originCommonRules into destinationCommonRules by id
+  mergeCommonRules(originCommonRules, destinationCommonRules) {
+    const destinationCommonRulesMap = new Map(destinationCommonRules.map((rule) => [rule.id, rule]))
+
+    originCommonRules.forEach((originRule) => {
+      if (destinationCommonRulesMap.has(originRule.id)) {
+        const destinationRuleIndex = destinationCommonRules.findIndex((r) => r.id === originRule.id)
```

Comment:
This code seems that it can be simplified, you don't need to use the findIndex function if you work with the map. In the map you have direct access to the rule that you want. In the end you can transform the map in an array.